CPDShell/Core/algorithms/DensityBasedCPD/abstracts/density_based_algorithm.py

```diff
@@ -0,0 +1,86 @@
+from abc import abstractmethod
+from collections.abc import Callable, Iterable
+
+import numpy as np
+from scipy.optimize import minimize
+from sklearn.neighbors import KernelDensity
+
+from CPDShell.Core.algorithms.abstract_algorithm import Algorithm
+
+
+class DensityBasedAlgorithm(Algorithm):
+    @staticmethod
+    def _kernel_density_estimation(observation: np.ndarray, bandwidth: float) -> KernelDensity:
+        """Perform kernel density estimation on the given observations.
+
+        Args:
+            observation (np.ndarray): the data points for which to estimate the density.
+            bandwidth (float): the bandwidth parameter for the kernel density estimation.
+
+        Returns:
+            KernelDensity: a fitted KernelDensity object.
+        """
+        kde = KernelDensity(kernel="gaussian", bandwidth=bandwidth).fit(observation)
```

Comment:
IDE says on `return kde`: "Expected type 'KernelDensity', got 'object' instead". It seems that you should separately + consequentially declare `kde` and use `.fit` on it (but I'm not sure), because `.fit` returns `object`.